src/components/AllowanceInput.vue

```diff
@@ -0,0 +1,168 @@
+<template>
+  <div class="amountInput allowance" :class="{ error: error }">
+    <i-input ref="amountInput" v-model="inputtedAmount" maxlength="35" size="lg" placeholder="Unlimited" type="text" @keyup.enter="$emit('enter')"></i-input>
+    <div class="error">
+      {{ error }}
+    </div>
+    <div v-if="token" class="_display-flex _justify-content-space-between">
+      <div class="secondaryText">
+        <!-- {{ inputtedAmountBigNumber | formatUsdAmount(token.tokenPrice, token.symbol) }} -->
+      </div>
+      <div class="linkText minAmount" @click="chooseMinAmount()">
+        <transition name="fadeFast">
+          <span v-if="minAmount !== '0'">Min: {{ minAmount | formatToken(token.symbol) }}</span>
+        </transition>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import utils from "@/plugins/utils";
+import { BigNumber } from "ethers";
+import Vue from "vue";
+
+export default Vue.extend({
+  props: {
+    value: {
+      type: String,
+      default: "",
+      required: false,
+    },
+    token: {
+      type: Object,
```

Comment:
I don't think you can apply TypeScript types to vue component props.

Reply:
At least you can't do it because of the way props are currently implemented. We are currently searching for a way how we can set specific TypeScript classes.